mobilecoind/src/service.rs

```diff
@@ -1210,6 +1219,14 @@ impl<T: BlockchainConnection + UserTxConnection + 'static> ServiceApi<T> {
                     mc_mobilecoind_api::ProcessedTxOutDirection::from_i32(src.direction)
                         .unwrap_or(mc_mobilecoind_api::ProcessedTxOutDirection::Invalid),
                 );
+
+                let subaddress = account_key.subaddress(src.subaddress_index);
+                let mut wrapper = mc_mobilecoind_api::printable::PrintableWrapper::new();
+                wrapper.set_public_address((&subaddress).into());
+                let encoded = wrapper
+                    .b58_encode()
+                    .unwrap_or_else(|_| "invalid address encoding".to_owned());
```

Comment:
This should return an RPC error, not end up as the value of address_code when encoding fails for whatever reason.

Reply:
I actually don't know how to do this. I looked at some options like `itertools::fold_results` and [`flat_map`](https://stackoverflow.com/questions/59852161/how-to-handle-result-in-flat-map).

But since we're constructing the encoding from data we collect internally rather than the user's data, I'm not sure it can ever fail anyway?

Reply:
The `.map()` code can be changed to return a Result. Then this call becames `.map_err(|e| format!("Failed b58 encoding of {:?}: {}", wrapper, e))?` and then end changes from `dst` to `Ok(dst)`. 
Then the `.collect()` call should change to `.collect::<Result<Vec<_>, String>>()?`, and I believe that should do it.